### PR TITLE
Iterate applications & use cache for syncing

### PR DESCRIPTION
**Pull request by fomentia**

### CODE REVIEW

**Comment by fomentia** [comment]
Previous: https://github.com/JupiterOne/graph-okta/pull/62
Next: none, thank you for making it here. :)

File: src/synchronizers/synchronizeApplications.ts [hunk 1]
```diff
@@ -54,40 +57,29 @@ export default async function synchronizeApplications(
...
-    await retryIfRateLimited(logger, () =>
-      applicationGroups.each((group: OktaApplicationGroup) => {
-        newApplicationGroupAndGroupRoleRelationships.push(
-          ...createApplicationGroupRelationships(applicationEntity, group),
-        );
-      }),
+  applicationsCache.forEach(entry => {
+    const applicationEntity = createApplicationEntity(
+      instance,
+      entry.data!.application,
```
Comment (aiwilliams) [inline src/synchronizers/synchronizeApplications.ts]: Shouldn't need the `!` on `entry.data`, FYI.

**Comment by fomentia** [comment]
Thought about this a bit more... I'll need to make some changes to the nested iteration of groups and users.

Basically, we loop through all the users of applications as part of the pagination of applications. If the organization has hundreds or thousands of users, though, and one application is assigned to all of them, we'll have some issues with that approach.

What I need to do to make this work more robustly is put application user iteration into its own step, similar to how we paginate over projects' issues in the Jira integration.

**Comment by fomentia** [comment]
OK, this should be good to go now. Seeing correct entity and relationship counts after some bug fixes.

**Pull request merged**